File: graph/connectedcomponent.java
import java.util.*;
class connectedcomponent{
    static class edge{
        int src; // source 
        int nbr; // neoubor
        int wt;  // weight
        edge(int v1,int v2, int wt){
            this.src=v1;
            this.nbr=v2;
            this.wt=wt;
        }
    }
   public static void addedge(ArrayList<ArrayList<edge>> graph, int src , int nbr,int wt){
     
    graph.get(src).add(new edge(src,nbr,wt));
    graph.get(nbr).add(new edge(nbr,src,wt));
   }

   public static void getcomponent(ArrayList<ArrayList<edge>> graph,ArrayList<Integer> comp,boolean[] visited,int s){
       visited[s] = true;
       comp.add(s);
       for(int e=0;e<graph.get(s).size();e++){
           edge ce=graph.get(s).get(e);
           if(visited[ce.nbr] == false){
               getcomponent(graph, comp, visited, ce.nbr);
           }
       }
   }
   public static int contcomp(ArrayList<ArrayList<edge>> graph){
       boolean[] visited=new boolean[graph.size()];
       int count=0;
       for(int v=0;v<graph.size();v++){
           if(visited[v] == false){
               count++;
               ArrayList<Integer> singlecomp=new ArrayList<>();
               getcomponent(graph,singlecomp,visited,v);
               System.out.println(singlecomp);
           }
       }
      return count;
  }

    public static void main(String[] args) {
        ArrayList<ArrayList<edge>> graph= new ArrayList<>();
    for(int i=0;i<=6;i++){
        graph.add(new ArrayList<edge>());
    }
    addedge(graph,0,1,10);
    addedge(graph,0,3,40);
    addedge(graph,1,2,10);
    addedge(graph,2,3,10);
    // addedge(graph,2,5,20); // for hamiltanion cycle 
    // addedge(graph,3,4,2); // for this question
    addedge(graph,4,5,3);
    addedge(graph,5,6,3);
    addedge(graph,4,6,8);
    // boolean[] visited =new boolean[graph.size()];
   int comp= contcomp(graph);
   System.out.print(comp);

    }
}

File: graph/ConnectedComponent.cpp
#include<iostream>
#include<vector>
using namespace std;
class Edge{
    public:
    int v , w = 0 ;
    Edge(int v , int w){
        this->v = v ;
        this->w = w ; 
    }
};

int n = 8;
 vector<vector<Edge>> graph(n, vector<Edge>());
void addEdge(vector<vector<Edge>> &gp , int v ,int u , int w){
    gp[u].push_back( Edge(v,w));
     gp[v].push_back( Edge(u,w));
}
void display(vector<vector<Edge>> &gp){
     for(int i = 0 ; i < gp.size() ; i++){
         cout<<i<<"->";
         for(Edge e : gp[i]){
             cout<<"(" << e.v <<","<< e.w <<"),";
         }
         cout<<endl;
     }
    cout<<endl;
 }
int Gcc_DFS(int src ,vector<bool> &vis){
    vis[src] =true;
    int count = 0 ;
    for(Edge e : graph[src]){
        if(!vis[e.v]){
            count+=Gcc_DFS(e.v , vis);
        }
    }
    return count+1;
}

 int Gcc(){
    vector<bool> vis(n, false);
    int count= 0; 
    int maxSize = 0;
    for(int i = 0;i < n; i++){
        if(!vis[i]){
            count++;
            maxSize = max(maxSize  , Gcc_DFS(i, vis));
        }
    }
    cout<<maxSize<<endl;
    return count;
}
 void constructgraph(){

    // for(int i=0;i<N;i++){
    //     vector<Edge*> a;
    //      graph.push_back(a);
    //  }
    addEdge(graph, 0, 1, 10);
    addEdge(graph, 0, 3, 10);
    addEdge(graph, 1, 2, 10);
    addEdge(graph, 2, 3, 40);
    // addEdge(graph, 3, 4, 2);
    addEdge(graph, 4, 5, 2);
    addEdge(graph, 4, 6, 3);
    addEdge(graph, 5, 6, 8);
    // addEdge(graph, 2,5,2);
    display(graph);
    cout<<endl;
}


int main(){
    constructgraph();
    cout<<Gcc()<<endl;
}